File: RecyclerViewGrids/app/src/main/java/com/example/recyclerviewgrids/ViewPagerAdapter.java
package com.example.recyclerviewgrids;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import androidx.fragment.app.FragmentStatePagerAdapter;

import com.example.recyclerviewgrids.Demo1.Demo1Fragment;
import com.example.recyclerviewgrids.Demo2.Demo2Fragment;
import com.example.recyclerviewgrids.Demo3.Demo3Fragment;

public class ViewPagerAdapter extends FragmentStatePagerAdapter {

    public ViewPagerAdapter(@NonNull FragmentManager fm, int behavior) {
        super(fm, behavior);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                return new Demo1Fragment();
            case 1:
                return new Demo2Fragment();
            case 2:
                return new Demo3Fragment();

        }
        return null;
    }

    @Override
    public int getCount() {
        return 3;
    }
}

File: Head-Tracking/app/src/main/java/com/black/xperiments/headtracking/views/ViewPagerAdapter.kt
package com.black.xperiments.headtracking.views

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.black.xperiments.headtracking.views.eye.EyeTrackingFragment
import com.black.xperiments.headtracking.views.head.HeadTrackingFragment
import com.black.xperiments.headtracking.views.head3d.Head3DFragment
import com.black.xperiments.headtracking.views.headnav.HeadNavigationFragment

class ViewPagerAdapter(manager: FragmentManager): FragmentStatePagerAdapter(
    manager,
    BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT
){
    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> EyeTrackingFragment.newInstance()
            1 -> HeadTrackingFragment.newInstance()
            2 -> HeadNavigationFragment.newInstance()
            3 -> Head3DFragment.newInstance()
            else -> EyeTrackingFragment.newInstance()
        }
    }
    override fun getCount() = 4
}